p2p/host/autorelay/relay_finder.go

```diff
@@ -222,14 +234,6 @@ func (rf *relayFinder) runScheduledWork(ctx context.Context, now time.Time, sche
 		scheduledWork.nextOldCandidateCheck = rf.clearOldCandidates(now)
 	}
 
-	if now.After(scheduledWork.nextAllowedCallToPeerSource) {
-		scheduledWork.nextAllowedCallToPeerSource = scheduledWork.nextAllowedCallToPeerSource.Add(rf.conf.minInterval)
-		select {
-		case peerSourceRateLimiter <- struct{}{}:
```

Comment:
Instead of these changes, we should simply set the `scheduledWork.nextAllowedCallToPeerSource` when we put a struct on this channel. I think that will have the same effect while being a smaller change and not introduce anything new.

Reply:
I guess it's not exactly the same, we still need to know when the peer source fn was called.

It might make sense for the findNodes method to signal via the new channel [here](https://github.com/libp2p/go-libp2p/pull/2187/files#diff-4fabce8a14510095c124d7159db6ae68844f99c52092ee12b05c1334323fa741R94) that we should run the background tasks.

Reply:
yeah the problem is knowing when peerSource was called. I'll check the changes happening in marco/test-max-age-flaky